src/status_im/wallet/core.cljs

```diff
@@ -780,15 +783,34 @@
             (navigation/navigate-back)))
 
 (fx/defn stop-fetching-on-empty-tx-history
-  [{:keys [db] :as cofx} transfers]
+  [{:keys [db now] :as cofx} transfers]
   (let [non-empty-history?    (get db :wallet/non-empty-tx-history?)
         custom-node?          (ethereum/custom-rpc-node?
-                               (ethereum/current-network db))]
-    (if (and (not non-empty-history?)
-             (empty? transfers)
-             (not custom-node?))
-      (clear-timeouts cofx)
-      {:db (assoc db :wallet/non-empty-tx-history? true)})))
+                               (ethereum/current-network db))
+        until-ms              (get db :wallet/keep-watching-until-ms)]
+    (when-not (and until-ms (> until-ms now))
+      (fx/merge
+       cofx
+       {:db (dissoc db :wallet/keep-watching-until-ms)}
+       (if (and (not non-empty-history?)
+                (empty? transfers)
+                (not custom-node?))
+         (clear-timeouts)
+         {:db (assoc db :wallet/non-empty-tx-history? true)})))))
```

Comment:
this won't work without you fix for fx/merge ?

Reply:
nope, this just won't work :) fixing

Reply:
The thing is that `fx/merge` will just pass `cofx` to that map and this will return `nil` with no effects as a result.

Reply:
fixed